tests/integration_tests/test_wandb.py

```diff
@@ -28,114 +30,267 @@ def _multiobjective_func(trial: optuna.trial.Trial) -> Tuple[float, float]:
+    wandb.init.assert_called_once_with(
+        project="optuna", group="summary", job_type="logging", mode="offline", tags=["test-tag"]
+    )
+
+    wandbc = WeightsAndBiasesCallback(
+        metric_name="mse", wandb_kwargs=wandb_kwargs, as_multirun=True
     )
+    wandb.run = None
+
+    study = optuna.create_study(direction="minimize")
+    _wrapped_func = wandbc.track_in_wandb(lambda t: 1.0)
+    wandb.init.reset_mock()
+    trial = optuna.create_trial(value=1.0)
+    _wrapped_func(trial)
 
     wandb.init.assert_called_once_with(
-        project="optuna",
-        group="summary",
-        job_type="logging",
-        mode="offline",
+        project="optuna", group="summary", job_type="logging", mode="offline", tags=["test-tag"]
+    )
+
+    wandb.init.reset_mock()
+    study.optimize(_objective_func, n_trials=n_trials, callbacks=[wandbc])
+
+    wandb.init.assert_called_with(
+        project="optuna", group="summary", job_type="logging", mode="offline", tags=["test-tag"]
     )
 
+    assert wandb.init.call_count == n_trials
+
+    wandb.init().finish.assert_called()
+    assert wandb.init().finish.call_count == n_trials
+
 
 @mock.patch("optuna.integration.wandb.wandb")
-def test_attributes_set_on_epoch(wandb: mock.MagicMock) -> None:
+@pytest.mark.parametrize("as_multirun", [True, False])
+def test_attributes_set_on_epoch(wandb: mock.MagicMock, as_multirun: bool) -> None:
 
-    wandb.config.update = mock.MagicMock()
+    # Vanilla update
+    wandb.sdk.wandb_run.Run = mock.MagicMock
+    expected = {"direction": ["MINIMIZE"]}
+    trial_params = {"x": 1.1, "y": 2.2}
+    expected_with_params = {"direction": ["MINIMIZE"], "x": 1.1, "y": 2.2}
 
-    wandbc = WeightsAndBiasesCallback()
     study = optuna.create_study(direction="minimize")
+    wandbc = WeightsAndBiasesCallback(as_multirun=as_multirun)
+    study.enqueue_trial(trial_params)
     study.optimize(_objective_func, n_trials=1, callbacks=[wandbc])
 
-    expected = {"direction": ["MINIMIZE"]}
-    wandb.config.update.assert_called_once_with(expected)
+    if as_multirun:
+        wandb.run = None
+        study.enqueue_trial(trial_params)
+        study.optimize(_objective_func, n_trials=1, callbacks=[wandbc])
+        wandb.init().config.update.assert_called_once_with(expected_with_params)
+    else:
+        wandb.run.config.update.assert_called_once_with(expected)
 
 
 @mock.patch("optuna.integration.wandb.wandb")
-def test_multiobjective_attributes_set_on_epoch(wandb: mock.MagicMock) -> None:
+@pytest.mark.parametrize("as_multirun", [True, False])
+def test_multiobjective_attributes_set_on_epoch(wandb: mock.MagicMock, as_multirun: bool) -> None:
 
-    wandb.config.update = mock.MagicMock()
+    wandb.sdk.wandb_run.Run = mock.MagicMock
+    trial_params = {"x": 1.1, "y": 2.2}
+    expected = {"direction": ["MINIMIZE", "MAXIMIZE"]}
+    expected_with_params = {"direction": ["MINIMIZE", "MAXIMIZE"], "x": 1.1, "y": 2.2}
```

Comment:
Unify the definition order and informative names.
```suggestion
    expected_config = {"direction": ["MINIMIZE", "MAXIMIZE"]}
    trial_params = {"x": 1.1, "y": 2.2}    
    expected_config_with_params = expected_config | trial_params
```